jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandDialog.kt

```diff
@@ -0,0 +1,101 @@
+import software.aws.toolkits.jetbrains.utils.ui.selected
+import software.aws.toolkits.resources.message
+import javax.swing.DefaultComboBoxModel
+import javax.swing.JComponent
+
+class RunCommandDialog(private val project: Project, private val container: ContainerDetails) : DialogWrapper(project) {
+    private val resourceCache = AwsResourceCache.getInstance()
+    private val tasks = resourceCache.getResourceNow(
+        EcsResources.listTasks(
+            container.service.clusterArn(),
+            container.service.serviceArn()
+        ),
+        project.activeRegion(),
+        project.activeCredentialProvider(),
+        useStale = false,
+        forceFetch = true
+    )
+
+    private val taskList = DefaultComboBoxModel(tasks.toTypedArray())
+    var task = if (tasks.isNotEmpty()) tasks.first() else null
+    private val command = JBTextField()
+    private val component by lazy {
+        panel {
+            row(message("ecs.execute_command_task")) {
+                comboBox(taskList, { task }, { if (it != null) { task = it } })
+                    .constraints(growX)
+                    .withErrorOnApplyIf(message("ecs.execute_command_task_comboBox_empty")) { it.selected() == null }
+            }
+            row(message("ecs.execute_command_label")) {
+                command(grow).withErrorOnApplyIf(message("ecs.execute_command_no_command")) { it.text == "" }
+            }
+        }
+    }
+    init {
+        super.init()
+        title = message("ecs.execute_command_run")
+        setOKButtonText(message("general.execute_button"))
+    }
+    override fun createCenterPanel(): JComponent? = component
+
+    override fun doOKAction() {
+        super.doOKAction()
+        runCommand()
+    }
+
+    fun constructExecCommandParameters(commandToExecute: String) = task?.let {
+        message(
+            "ecs.execute_command_parameters",
+            container.service.clusterArn(),
+            it,
+            commandToExecute
+        )
+    }
+
+    private fun runCommand() {
+        val execCommand = Tool()
+        updateExecCommandRunSettings(execCommand, command.text)
+        val environment = ExecutionEnvironmentBuilder
+            .create(
+                project,
+                DefaultRunExecutor.getRunExecutorInstance(),
+                ToolRunProfile(
+                    execCommand,
+                    SimpleDataContext.getProjectContext(project)
+                )
+            )
+            .build()
+        environment.runner.execute(environment)
+    }
+
+    fun updateExecCommandRunSettings(execCommand: Tool, commandToExecute: String) {
+        execCommand.isShowConsoleOnStdOut = true
+        execCommand.isUseConsole = true
+        execCommand.parameters = constructExecCommandParameters(commandToExecute)
+        execCommand.name = container.containerDefinition.name()
+        execCommand.program = path
+    }
+
+    companion object {
+        var path = ""
```

Comment:
What is this static state doing here?

Reply:
This is the path for AWS CLI executable retrieved after it is validated. it is used as a parameter for the tool